sklearn/metrics/scorer.py

```diff
@@ -37,10 +38,15 @@ def __init__(self, score_func, sign, kwargs):
         self._kwargs = kwargs
         self._score_func = score_func
         self._sign = sign
+        # XXX After removing the deprecated scorers (v0.20) remove the
+        # XXX deprecation_msg property again and make __call__ abstract again
+        self._deprecation_msg = None
 
-    @abstractmethod
     def __call__(self, estimator, X, y, sample_weight=None):
-        pass
+        if self._deprecation_msg is not None:
+            warnings.warn(self._deprecation_msg,
+                          category=DeprecationWarning,
+                          stacklevel=13)
```

Comment:
What is a good default then? If you use `cross_val_score()` you need stackelvel=13 because of all the `Parallel` stuff. So I have no opinion really on what to set it to. It seems like getting the warning to actually point to where you made the choice that leads to the warning is nearly impossible??